pandas/core/arrays/sparse.py

```diff
@@ -1471,15 +1471,32 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
             'power': 'pow',
             'remainder': 'mod',
             'divide': 'div',
+            'equal': 'eq',
+            'not_equal': 'ne',
+            'less': 'lt',
+            'less_equal': 'le',
+            'greater': 'gt',
+            'greater_equal': 'ge',
         }
+
+        flipped = {
+            'lt': '__gt__',
+            'le': '__ge__',
+            'gt': '__lt__',
+            'ge': '__le__',
+            'eq': '__eq__',
+            'ne': '__ne__',
+        }
+
         op_name = ufunc.__name__
         op_name = aliases.get(op_name, op_name)
 
         if op_name in special and kwargs.get('out') is None:
             if isinstance(inputs[0], type(self)):
                 return getattr(self, '__{}__'.format(op_name))(inputs[1])
             else:
-                return getattr(self, '__r{}__'.format(op_name))(inputs[0])
+                name = flipped.get(op_name, '__r{}__'.format(op_name))
```

Comment:
note to @jbrockmendel since we do this a couple of times iIRC, we should have a more generic way of doing this

Reply:
And if we implement `__array_ufunc__` on more arrays, we'll need to do it in those places too.

I think `pandas.core.ops._op_descriptions` may have enough info.

Reply:
That doesn't quite work since the comparison ops don't define `reversed`, which may be sensible (haven't really thought it through).